### ISSUE TITLE
Get grad of inputs

### ISSUE THREAD

**Issue by xiezouqu** [fredzzhang/pvic#55]
Hello, I respectfully appreciate the work you have done. I encountered the following issue during getting grad of inputs, and I would greatly appreciate your help in solving it.

function  def _on_each_iteration(self):  in  util.py 

```
self._state.loss = sum(loss for loss in loss_dict.values())
        self._state.optimizer.zero_grad(set_to_none=True)
        self._state.loss.backward()

        #To get inputs.grad
        grad = self._state.inputs[0][0].grad
```
but grad is None, I want to know how to get the correct the grad.and I would greatly appreciate your help in solving it. 
@fredzzhang

**Comment by fredzzhang** [comment]
Hi @xiezouqu,

The `tensor.grad` attribute only works when the tensor has `requires_grad` set to `True`. Normally it's only applied to the network parameters that will be updated. If you want to get the gradient of the parameters, you need to call `.grad` on those parameters.

Fred.

**Comment by xiezouqu** [comment]
Hi @fredzzhang 
Thank you very much for your response！
